x-pack/test/functional/services/monitoring/no_data.js

```diff
@@ -4,13 +4,20 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export function MonitoringNoDataProvider({ getService }) {
+export function MonitoringNoDataProvider({ getService, getPageObjects }) {
   const testSubjects = getService('testSubjects');
   const retry = getService('retry');
+  const PageObjects = getPageObjects(['common']);
 
   return new class NoData {
     async enableMonitoring() {
-      await testSubjects.click('useInternalCollection');
+      // Cloud currently does not support Metricbeat-based collection
+      // so the UI does not give the user a choice between the two collection
+      // methods. So if we're on cloud, do not try and switch to internal collection
+      // as it's already the default
+      if (!await PageObjects.common.isCloud()) {
+        await testSubjects.click('useInternalCollection');
```

Comment:
does there need to be a `return` here?

Reply:
Nah. For cloud, there isn't a button that says `Use internal collection` (since that's the only option). On prem, we default to using metricbeat collection so there is a button that allows the user to use internal collection. For this test, we just want to setup internal collection so for on prem, we need to click that button which allows us to click the `enableCollectionEnabled` button. For cloud, the `enableCollectionEnabled` is there by default

Reply:
Thanks, I get it now. For non-cloud, both links need clicking. But for cloud only the latter link does